api/incubator/src/main/java/io/opentelemetry/api/incubator/config/internal/YamlStructuredConfigProperties.java

```diff
@@ -26,41 +25,45 @@
  * Implementation of {@link StructuredConfigProperties} which uses a file configuration model as a
  * source.
  *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ *
  * @see #getStructured(String) Accessing nested maps
  * @see #getStructuredList(String) Accessing lists of maps
- * @see FileConfiguration#toConfigProperties(Object) Converting configuration model to properties
  */
-final class YamlStructuredConfigProperties implements StructuredConfigProperties {
+public final class YamlStructuredConfigProperties implements StructuredConfigProperties {
 
   /** Values are {@link #isPrimitive(Object)}, {@link List} of scalars. */
   private final Map<String, Object> simpleEntries;
 
   private final Map<String, List<StructuredConfigProperties>> listEntries;
   private final Map<String, StructuredConfigProperties> mapEntries;
+  private final Set<String> nullKeys;
 
   private YamlStructuredConfigProperties(
       Map<String, Object> simpleEntries,
       Map<String, List<StructuredConfigProperties>> listEntries,
-      Map<String, StructuredConfigProperties> mapEntries) {
+      Map<String, StructuredConfigProperties> mapEntries,
+      Set<String> nullKeys) {
     this.simpleEntries = simpleEntries;
     this.listEntries = listEntries;
     this.mapEntries = mapEntries;
+    this.nullKeys = nullKeys;
   }
 
   /**
    * Create a {@link YamlStructuredConfigProperties} from the {@code properties} map.
    *
-   * <p>{@code properties} is expected to be the output of YAML parsing (i.e. with Jackson {@link
+   * <p>{@code properties} is expected to be the output of YAML parsing (i.e. with Jackson {@code
    * com.fasterxml.jackson.databind.ObjectMapper}), and have values which are scalars, lists of
    * scalars, lists of maps, and maps.
-   *
-   * @see FileConfiguration#toConfigProperties(OpenTelemetryConfiguration)
    */
   @SuppressWarnings("unchecked")
-  static YamlStructuredConfigProperties create(Map<String, Object> properties) {
+  public static YamlStructuredConfigProperties create(Map<String, Object> properties) {
     Map<String, Object> simpleEntries = new HashMap<>();
     Map<String, List<StructuredConfigProperties>> listEntries = new HashMap<>();
     Map<String, StructuredConfigProperties> mapEntries = new HashMap<>();
+    Set<String> nullKeys = new HashSet<>();
```

Comment:
only other time I'll mentioned LinkedHashSet as I assume this is a project norm and if we wanted to have consistent iteration at low cost, we wouldn't just do LHM LHS here, but everywhere